### PR TITLE
SDL_LoadBMP: Copied documentation from old wiki and added an example

Live page is here: https://wiki.libsdl.org/SDL_LoadBMP



### PR DESCRIPTION
**Pull request by SDLWikiBot**
This edit was made by @josiest.

Live page is here: https://wiki.libsdl.org/SDL_LoadBMP

If this user should be blocked from further edits, an admin should go to https://wiki.libsdl.org/josiest/block
If this user should be trusted to make direct pushes to main, without a pull request, an admin should go to https://wiki.libsdl.org/josiest/trust

### CODE REVIEW

**Comment by josiest** [comment]
I'm not sure if `<<Include(SDL_LoadBMP_RW, , , from="== Return Value ==", to="== Code Examples ==")>>` is supposed to be a placeholder, or if it's supposed to actually load information from the `SDL_LoadBMP_RW`. If it's the latter, it doesn't work on the browsers I've tried (chrome 94.0.4606.54 and firefox 78.14.0esr). I've sug

File: SDL_LoadBMP.mediawiki [hunk 1]
```diff
@@ -15,10 +14,31 @@ SDL_Surface* SDL_LoadBMP(const char* file)
...
 
 == Remarks ==
-<<Include(SDL_LoadBMP_RW, , , from="== Remarks ==", to="== Related Functions ==")>>
+The new surface should be freed with [[SDL_FreeSurface]]().
+
+== Code Example ==
+
+<syntaxhighlight lang='c++'>
+// try to load the bitmap image
+const char * image_path = "path/to/image.bmp";
```
Comment (josiest) [inline SDL_LoadBMP.mediawiki:26]: Whoops! I've been working with C++ lately and forgot how to declare a c-string. This should be `const char image_path[] = "path/to/image.bmp";`

**Comment by josiest** [comment]
Also, I'm not entirely sure if this is the best example. While the if-else illustrates that the the surface resource needs to be released, this example implies a heavily nested conditional structure - which I'm not a big fan of and don't really want to encourage. Maybe a better example could be something like

```c
int main(int argc, char * argv[])
{
    // set up
    ...
    // try to load the image
    const char * image_path = "path/to/image.bmp"
    SDL_Surface * surface = SDL_LoadBMP(image_path);
    
    // let the user know if the image failed to load
    if (NULL == surface) {
        printf("Couldn't load image at %s: %s\n", image_path, SDL_GetError());
        // clean up
        ...
        return -1;
    }
    // do stuff with surface
    ...
    // finally, make sure to release the surface resource
    SDL_FreeSurface(surface);
    // clean up
    ...
}
```

**Comment by icculus** [comment]
> I'm not sure if <<Include(SDL_LoadBMP_RW, , , from="== Return Value ==", to="== Code Examples ==")>> is supposed to be a placeholder

It's a leftover from when we were running MoinMoin; it would insert text from another page on the wiki. The new software doesn't support it and it always seemed fragile to me anyhow.

I'm wondering if SDL_LoadBMP's page should just say something like "this is a macro to make SDL_LoadBMP_RW easier" and link to that page (and maybe move the code example there if there isn't one already).

**Comment by josiest** [comment]
> I'm wondering if SDL_LoadBMP's page should just say something like "this is a macro to make SDL_LoadBMP_RW easier" and link to that page (and maybe move the code example there if there isn't one already).

Oh yeah, I should have thought of that! I actually ran into an issue with that the other day when I tried to use the address of `SDL_LoadBMP` as a function pointer. I'm not super familiar with how the wiki edits work though: is it possible to make changes to this branch, or would I need to make another pull request?

**Comment by icculus** [comment]
Just make a new edit on the wiki, I'll close this one.

**Pull request closed without merging**